rekall-core/rekall/plugins/linux/find_kaslr.py

```diff
@@ -0,0 +1,202 @@
+#       SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+#       CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+#       OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+#       OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+"""
+Mainly based on the great blogpost series by bneuburg:
+    https://bneuburg.github.io/volatility/kaslr/2017/04/26/KASLR1.html
+    https://bneuburg.github.io/volatility/kaslr/2017/05/05/KASLR2.html
+    https://bneuburg.github.io/volatility/kaslr/2017/05/16/KASLR3.html
+and on the modifications included in Volatility:
+    https://github.com/volatilityfoundation/volatility/commit/3619c8a32f5b2f552bbc4a8464a6ea768de48423
+
+Tested on x86 and x86_64 for NOKASLR and KASLR 1st and 2nd generation.
+"""
+
+__author__ = "Frank Block <fblock@ernw.de>"
+
+from rekall import scan
+from rekall.plugins.linux import common
+import struct
+
+
+class SwapperScanner(scan.BaseScanner):
+
+    checks = [("StringCheck", dict(needle=b"swapper/0\x00"))]
+
+
+class FindKaslr(common.AbstractLinuxCommandPlugin):
+
+    name = "find_kaslr"
+
+    __args = [
+        dict(name='scan_whole_physical_space', type="Boolean", default=False,
+             help='Normally, we scan at max only the first gigabyte of '
+                  'memory, which should be fine as the linux kernel and its '
+                  'data is normally located in the beginning of the physical '
+                  'memory space.')
+    ]
+
+    table_header = [
+        dict(name="physical_shift", width=16, style="address"),
+        dict(name="virtual_shift", width=16, style="address"),
+        dict(name="kernel_slide", width=16, style="address"),
+        dict(name="page_offset", width=16, style="address"),
+        dict(name="DTB", width=16, style="address"),
+        dict(name="Valid", width=16),
+        dict(name="comment", width=48)
+    ]
+
+    def collect(self):
+        shifts  = [0xffffffff80000000]
+        rekall_shifts  = [0xffff80000000]
+        if self.session.profile.metadata("arch") == 'AMD64':
+            pass
+
+        elif self.session.profile.metadata("arch") == 'I386':
+            shifts = [0xc0000000]
+            rekall_shifts = shifts
+
+        else:
+            self.session.error.logging.error(
+                "Unsupported architecture.")
+            return
+
+        physical_shift = 0
+        virtual_shift = 0
+        kernel_slide = 0
+        page_offset = 0
+
+        comm_offset = self.profile.get_obj_offset('task_struct', 'comm')
+        pid_offset = self.profile.get_obj_offset('task_struct', 'pid')
+        files_offset = self.profile.get_obj_offset('task_struct', 'files')
+        pas = self.session.physical_address_space
+
+        pointer_size = self.profile.get_obj_size("Pointer")
+        fmt = 'I' if pointer_size == 4 else 'Q'
+
+        init_task_address = self.profile.get_constant('init_task',
```

Comment:
You probably wont need to mess with the different shift if you just used obj.Pointer.integer_to_address() here.

See this for example usage:
https://github.com/google/rekall/blob/8db49aa92439d4787023c1ea37561b6903cabf9b/rekall-core/rekall/plugins/linux/address_resolver.py#L53